test/parser_test.cpp

```diff
@@ -14,3 +16,39 @@ TEST(ParserTest, Empty) {
     StrictMock<MockEventHandler> handler;
     EXPECT_FALSE(parser.HandleNextDocument(handler));
 }
+
+TEST(ParserTest, CVE_2017_5950) {
+    std::string excessive_recursion(16384, '[');
+    std::istringstream input{excessive_recursion};
+    Parser parser{input};
+
+    NiceMock<MockEventHandler> handler;
+    EXPECT_THROW(parser.HandleNextDocument(handler), YAML::ParserException);
+}
+
+TEST(ParserTest, CVE_2018_20573) {
+    std::string excessive_recursion(20535, '{');
+    std::istringstream input{excessive_recursion};
+    Parser parser{input};
+
+    NiceMock<MockEventHandler> handler;
+    EXPECT_THROW(parser.HandleNextDocument(handler), YAML::ParserException);
+}
+
+TEST(ParserTest, CVE_2018_20574) {
+    std::string excessive_recursion(21989, '{');
```

Comment:
What's the difference between this and the one above, except the slightly different recursion depth?

Reply:
The CVEs differ by the recursion depths. (I know!)